Tribler/Test/test_my_channel.py

```diff
@@ -125,6 +129,88 @@ def do_page():
 
         self.startTest(do_page)
 
+    def test_metadata(self):
+        def take_screenshot_quit():
+            self.screenshot('Home')
+            self.quit()
+
+        def check_home():
+            self.CallConditional(20, lambda: len(self.guiUtility.frame.home.aw_panel.list.GetItems()) == 1,
+                                 take_screenshot_quit, "No thumbnail torrent in Home panel")
+
+        def on_channel_torrent_created(subject, events, object_id, data_list):
+            add_metadata = False
```

Comment:
@NielsZeilemaker I first came up with two ideas: one is to start a metadata-injector create a channel and some thumbnails, and then start a normal tribler, subscribe to that channel, and wait for the thumbnails to arrive. Another one is that one: I create a channel myself, add some thumbnails to it, and then I should be able to see them on the home page.

Here I took the second one. Because we no longer expose the metadata functionality to the users, in order to keep it simple, I have to do some of those crazy magic. Otherwise, a good way is to create an API wrapper for channels, but this requires a lot of refactory again.

The problem with the two instance approach is that, the metadata may take a long time to arrive on the normal tribler instance. In my case, when I manually tested the injector, it took at least 5 minutes for me to be able to see the channel which I create with the injector, and another 5 to 10 minutes to get the thumbnails. So, if we make the unit test this way, it might be a longest test ever.

As to the ffmpeg thing, since we are in full control of the metadata, there is no need for ffmpeg any more because we retrieve the thumbnails through certain TV or movie databases.